### ISSUE TITLE
No divider in callout

### ISSUE THREAD

**Issue by ZPhilo** [trevorsandy/lpub3d#306]
### Subject
If I right click on an assembly in a callout, the menu offers to add a divider, either horizontal or vertical. But if I accept, nothing happens, neither in callout nor in source code. 

### Environment
LPub3D - 2.3.12 / Win10

**Comment by trevorsandy** [comment]
Thank you for reporting this behaviour. I am not able to reproduce it with the latest available build. Perhaps you can provide an example model file that produces the behaviour on your system?

Here is a shot sequence and model file for the behaviour in question:

![Screenshot - 06_07_2019 , 07_50_40](https://user-images.githubusercontent.com/13388970/60752347-97c22400-9fc4-11e9-9b1b-0acc008cd0cb.png)

![Screenshot - 06_07_2019 , 07_50_56](https://user-images.githubusercontent.com/13388970/60752348-97c22400-9fc4-11e9-88ea-e6a63a712a9c.png)

![Screenshot - 06_07_2019 , 07_51_13](https://user-images.githubusercontent.com/13388970/60752349-97c22400-9fc4-11e9-99de-cd1804381c0b.png)

![Screenshot - 06_07_2019 , 07_51_50](https://user-images.githubusercontent.com/13388970/60752350-985aba80-9fc4-11e9-9c31-470df5f9c87f.png)

See #185 for details on the significance and use of selecting an Allocation other than the default.
![Screenshot - 06_07_2019 , 07_52_07](https://user-images.githubusercontent.com/13388970/60752351-985aba80-9fc4-11e9-9445-b415336793f1.png)

Note that when you select the context menu "Display Step as Rows" here, the divider will automatically be placed horizontally.
![Screenshot - 06_07_2019 , 07_52_24](https://user-images.githubusercontent.com/13388970/60752352-985aba80-9fc4-11e9-9937-b9031e8d3a3e.png)

A Vertical (Range - i.e. divide Steps in a range) divider is selected here for demonstration purposes only. Normally, the number of steps in this callout would make a 'Range' divider useless; however as you can see in the next shot, the divider is placed at the top of the callout.
![Screenshot - 06_07_2019 , 07_54_02](https://user-images.githubusercontent.com/13388970/60752353-985aba80-9fc4-11e9-83a4-55e2f5598cd5.png)

![Screenshot - 06_07_2019 , 07_54_42](https://user-images.githubusercontent.com/13388970/60752355-985aba80-9fc4-11e9-960e-4dcb3984aa1a.png)

Here are the before and after states of the example model file:

**Before**

```
0 FILE sample.ldr
0 sample
0 Name: sample.ldr
0 Author: foo
0 !LPUB SUBMODEL_DISPLAY SHOW_INSTANCE_COUNT GLOBAL TRUE
0 !LPUB SUBMODEL_DISPLAY SHOW GLOBAL TRUE
1 4 0 0 0 1 0 0 0 1 0 0 0 1 3001.dat
1 4 -40 24 0 1 0 0 0 1 0 0 0 1 3001.dat
1 4 40 24 0 1 0 0 0 1 0 0 0 1 3001.dat
0 ROTSTEP END
1 4 80 0 0 1 0 0 0 1 0 0 0 1 3001.dat
1 4 120 24 0 1 0 0 0 1 0 0 0 1 3001.dat
1 4 40 -24 0 1 0 0 0 1 0 0 0 1 3001.dat
0 STEP
1 1 -20 -8 0 1 0 0 0 1 0 0 0 1 subModel-1.ldr
0 STEP
1 1 -60 16 0 1 0 0 0 1 0 0 0 1 subModel-2.ldr
0 ROTSTEP 0 -90 0 REL
1 1 100 -8 0 1 0 0 0 1 0 0 0 1 subModel-2.ldr
0 ROTSTEP 0 180 0 REL
1 1 140 16 0 1 0 0 0 1 0 0 0 1 subModel-1.ldr
0 ROTSTEP 0 90 0 REL
0 STEP
0 NOFILE 
0 FILE submodel-1.ldr
0 Name: subModel-1.ldr
0 Author: foo
0 !LPUB MULTI_STEP BEGIN
1 1 0 0 0 1 0 0 0 1 0 0 0 1 87580.dat
0 ROTSTEP END
1 14 0 -24 0 1 0 0 0 1 0 0 0 1 3062b.dat
0 STEP
0 !LPUB MULTI_STEP END
0 NOFILE 
0 FILE submodel-2.ldr
0 Name: subModel-2.ldr
0 Author: foo
1 1 0 0 0 1 0 0 0 1 0 0 0 1 87580.dat
0 ROTSTEP END
1 25 0 -24 0 1 0 0 0 1 0 0 0 1 3062b.dat
0 STEP
0 NOFILE 

```

**After**

```
0 FILE sample.ldr
0 sample
0 Name: sample.ldr
0 Author: foo
0 !LPUB SUBMODEL_DISPLAY SHOW_INSTANCE_COUNT GLOBAL TRUE
0 !LPUB SUBMODEL_DISPLAY SHOW GLOBAL TRUE
1 4 0 0 0 1 0 0 0 1 0 0 0 1 3001.dat
1 4 -40 24 0 1 0 0 0 1 0 0 0 1 3001.dat
1 4 40 24 0 1 0 0 0 1 0 0 0 1 3001.dat
0 ROTSTEP END
1 4 80 0 0 1 0 0 0 1 0 0 0 1 3001.dat
1 4 120 24 0 1 0 0 0 1 0 0 0 1 3001.dat
1 4 40 -24 0 1 0 0 0 1 0 0 0 1 3001.dat
0 STEP
0 !LPUB CALLOUT BEGIN
1 1 -20 -8 0 1 0 0 0 1 0 0 0 1 subModel-1.ldr
0 !LPUB CALLOUT POINTER CENTER 0 0.253197 0.161765 0 0 0 0 0 0 1
0 !LPUB CALLOUT END
0 STEP
1 1 -60 16 0 1 0 0 0 1 0 0 0 1 subModel-2.ldr
0 ROTSTEP 0 -90 0 REL
1 1 100 -8 0 1 0 0 0 1 0 0 0 1 subModel-2.ldr
0 ROTSTEP 0 180 0 REL
1 1 140 16 0 1 0 0 0 1 0 0 0 1 subModel-1.ldr
0 ROTSTEP 0 90 0 REL
0 STEP
0 NOFILE 
0 FILE submodel-1.ldr
0 Name: subModel-1.ldr
0 Author: foo
1 1 0 0 0 1 0 0 0 1 0 0 0 1 87580.dat
0 ROTSTEP END
0 !LPUB CALLOUT DIVIDER 
1 14 0 -24 0 1 0 0 0 1 0 0 0 1 3062b.dat
0 STEP
0 NOFILE 
0 FILE submodel-2.ldr
0 Name: subModel-2.ldr
0 Author: foo
1 1 0 0 0 1 0 0 0 1 0 0 0 1 87580.dat
0 ROTSTEP END
1 25 0 -24 0 1 0 0 0 1 0 0 0 1 3062b.dat
0 STEP
0 NOFILE
```

Cheers,

**Comment by ZPhilo** [comment]
That's indeed the behaviour I expected... I'll try to see what triggers the problem on my file.

**Comment by ZPhilo** [comment]
I PMed you on LDraw forum with the file that causes problem. And I checked with your sample file, works perfectly here indeed.

**Comment by trevorsandy** [comment]
Got it, Thanks.  I was able to reproduce the this behaviour with your model file, it has been corrected for v2.3.13.

This behaviour occurs only under a specific condition; hence, why I did not reproduce it before. It appears when a divider is placed before the last step (which must have at least 1 type 1-5 part), for example...

```
0 Name: calledoutsubmodel.ldr
0 Author: LPub3D
1 4 0 24 0 1 0 0 0 1 0 0 0 1 3062b.dat
0 STEP
0 // Divider Meta command should be placed here but nothing happens!
1 4 0 0 0 1 0 0 0 1 0 0 0 1 4589.dat
```

... the current logic will look to find the boundary of _next step_ from the one containing the CSI item from which you launched the context menu to place the divider. The unexpected behaviour occurs when the end of the next step is the end of the submodel. 

### WORKAROUND
Place an empty step ment `0 STEP` after the last step like so...

```
0 Name: calledoutsubmodel.ldr
0 Author: LPub3D
1 4 0 24 0 1 0 0 0 1 0 0 0 1 3062b.dat
0 STEP
1 4 0 0 0 1 0 0 0 1 0 0 0 1 4589.dat
0 // Add an empty STEP at the end of the submodel to trigger expected behaviour
0 STEP
```

Cheers,
